#! /bin/sh /usr/share/dpatch/dpatch-run
##
## Patch by Bastian Kleineidam <calvin@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Handle the case of upgrading from Mailman 2.0 where we have
## DP: pending subscriptions.


@DPATCH@
diff -urNad trunk/bin/update /tmp/dpep.sAkJKA/trunk/bin/update
--- trunk/bin/update	2005-04-14 15:30:00.858629713 +0200
+++ /tmp/dpep.sAkJKA/trunk/bin/update	2005-04-14 15:32:19.483231517 +0200
@@ -505,9 +505,11 @@
     file20 = os.path.join(mm_cfg.DATA_DIR, 'pending_subscriptions.db')
     file214 = os.path.join(mm_cfg.DATA_DIR, 'pending.pck')
     db = None
+    ver = None
     # Try to load the Mailman 2.0 file
     try:
         fp = open(file20)
+        ver = "20"
     except IOError, e:
         if e.errno <> errno.ENOENT: raise
     else:
@@ -519,6 +521,7 @@
         # Try to load the Mailman 2.1.x where x < 5, file
         try:
             fp = open(file214)
+            ver = "214"
         except IOError, e:
             if e.errno <> errno.ENOENT: raise
         else:
@@ -552,8 +555,12 @@
             # data[0] is the address being unsubscribed
             addrops_by_address.setdefault(data[0], []).append((key, val))
         elif op == Pending.SUBSCRIPTION:
-            # data[0] is a UserDesc object
-            addr = data[0].address
+            if ver == "20":
+                # data is tuple (emailaddr, password, digest)
+                addr = data[0]
+            else:
+                # data[0] is a UserDesc object
+                addr = data[0].address
             subs_by_address.setdefault(addr, []).append((key, val))
         elif op == Pending.RE_ENABLE:
             # data[0] is the mailing list's internal name
